packages/scripts/src/admin-proposals/change-final-fee/0_Propose.ts

```diff
@@ -76,64 +74,93 @@ async function main() {
     const decimals = await erc20.decimals();
 
+    // Check if the token is already whitelisted
+    const addressWhitelist = await getConnectedAddressWhitelist(1);
+    const isWhitelisted = await addressWhitelist.isOnWhitelist(tokenAddress);
+    if (!isWhitelisted) {
+      console.log(`Adding ${token} to the whitelist on mainnet...`);
+      const addAddressTx = await addressWhitelist.populateTransaction.addToWhitelist(tokenAddress);
+      if (!addAddressTx.data) throw "addAddressTx.data is null";
+      adminProposalTransactions.push({ to: addressWhitelist.address, value: 0, data: addAddressTx.data });
+    }
+
+    // Check if the final fee is different from the current one
+    const store = await getConnectedStore(1);
+    const currentFinalFee = await store.finalFees(tokenAddress);
+    if (currentFinalFee.eq(hre.ethers.utils.parseUnits(newFinalFee.toString(), decimals))) {
+      console.log(`Final fee for ${token} is already ${newFinalFee} on mainnet`);
+      continue;
+    }
+    console.log(`Setting final fee for ${token} to ${newFinalFee} in mainnet...`);
     const setFinalFeeTx = await store.populateTransaction.setFinalFee(tokenAddress, {
       rawValue: hre.ethers.utils.parseUnits(newFinalFee.toString(), decimals),
     });
     if (!setFinalFeeTx.data) throw "setFinalFeeTx.data is null";
     adminProposalTransactions.push({ to: store.address, value: 0, data: setFinalFeeTx.data });
   }
 
-  for (const i in l2Networks) {
-    const networkName = l2Networks[i];
-    const provider = getRetryProvider(l2NetworksNumber[networkName as keyof typeof l2NetworksNumber]) as Provider;
-    const l2ChainId = l2NetworksNumber[networkName as keyof typeof l2NetworksNumber];
+  for (const networkName of supportedNetworks.filter((network) => network !== "mainnet")) {
+    if (!isSupportedNetwork(networkName)) throw new Error(`Unsupported network: ${networkName}`);
+    const l2ChainId = networksNumber[networkName];
+    const provider = getRetryProvider(l2ChainId) as Provider;
     const l2NodeUrl = process.env[String(NODE_URL_ENV + l2ChainId)];
     const isPolygon = l2ChainId === 137;
     const isArbitrum = l2ChainId === 42161;
 
-    // If is arbitrum we need to fund the parent messenger
-    if (isArbitrum)
-      await fundArbitrumParentMessengerForRelays(
-        arbitrumParentMessenger,
-        proposerSigner,
-        Object.keys(tokensToUpdateFee).length
-      );
-
     const governanceMessages: { targetAddress: string; tx: PopulatedTransaction }[] = [];
-
-    for (const token in tokensToUpdateFee) {
-      const isWeth = token == "WETH";
-      const newFinalFee = isWeth ? newFinalFeeWeth : newFinalFeeUSD;
-      const tokenAddresses = tokensToUpdateFee[token as keyof typeof tokensToUpdateFee];
-      const tokenAddress = tokenAddresses[networkName as keyof typeof tokenAddresses];
+    let fundArbitrumCount = 0;
+    for (const token in tokensToUpdate) {
```

Comment:
nit: prefer Object.entries when accessing the key and values. It's a little safer and requires less repeated access.

for (const [token, updateInfo] of Object.entries(tokensToUpdate)) {
  const newFinalFee = updateInfo.finalFee;
  const tokenAddress = updateInfo[networkName];